src/content/app/genome-browser/state/browser-sidebar-modal/browserSidebarModalSlice.ts

```diff
@@ -0,0 +1,160 @@
+ */
+
+import {
+  createSlice,
+  type PayloadAction,
+  type Action,
+  type ThunkAction
+} from '@reduxjs/toolkit';
+
+import { getBrowserActiveGenomeId } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSelectors';
+import { getActiveBrowserSidebarModal } from './browserSidebarModalSelectors';
+
+import type { ParsedUrlPayload } from 'src/content/app/genome-browser/state/browser-general/browserGeneralSlice';
+import type { RootState } from 'src/store';
+
+export enum BrowserSidebarModalView {
+  SEARCH = 'search',
+  TRACKS_MANAGER = 'Tracks manager',
+  BOOKMARKS = 'Previously viewed',
+  PERSONAL_DATA = 'Personal data',
+  SHARE = 'Share',
+  DOWNLOADS = 'Downloads'
+}
+
+export type BrowserSidebarModalStateForGenome = Readonly<{
+  browserSidebarModalView: BrowserSidebarModalView | null;
+}>;
+
+export type BrowserSidebarModalState = Readonly<{
+  [genomeId: string]: BrowserSidebarModalStateForGenome;
+}>;
+
+export const defaultBrowserSidebarModalStateForGenome: BrowserSidebarModalStateForGenome =
+  {
+    browserSidebarModalView: null
+  };
+
+export const changeBrowserSidebarModalViewForGenome =
+  (
+    browserSidebarModalView: BrowserSidebarModalView
+  ): ThunkAction<void, any, null, Action<string>> =>
+  (dispatch, getState: () => RootState) => {
+    const activeGenomeId = getBrowserActiveGenomeId(getState());
+
+    if (!activeGenomeId) {
+      return;
+    }
+
+    const data = {
+      ...getActiveBrowserSidebarModal(getState()),
+      browserSidebarModalView
+    };
+
+    dispatch(
+      updateBrowserSidebarModalForGenome({
+        activeGenomeId,
+        data
+      })
+    );
+  };
+
+export const openBrowserSidebarModal =
+  (
+    browserSidebarModalView: BrowserSidebarModalView
+  ): ThunkAction<void, any, null, Action<string>> =>
+  (dispatch, getState: () => RootState) => {
+    const state = getState();
+
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+
+    if (!activeGenomeId) {
+      return;
+    }
+
+    const data = {
+      ...getActiveBrowserSidebarModal(state),
+      browserSidebarModalView
+    };
+
+    dispatch(
+      updateBrowserSidebarModalForGenome({
+        activeGenomeId,
+        data
+      })
+    );
+  };
+
+export const closeBrowserSidebarModal =
+  (): ThunkAction<void, any, null, Action<string>> =>
+  (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const activeGenomeId = getBrowserActiveGenomeId(state);
+
+    if (!activeGenomeId) {
+      return;
+    }
+
+    const data = {
+      ...getActiveBrowserSidebarModal(state),
+      browserSidebarModalView: null
+    };
+
+    dispatch(
+      updateBrowserSidebarModalForGenome({
+        activeGenomeId,
+        data
+      })
+    );
+  };
+
+const browserSidebarModal = createSlice({
+  name: 'genome-browser-sidebar-modal',
+  initialState: {} as BrowserSidebarModalState,
+  reducers: {
+    setInitialBrowserSidebarModalDataForGenome(
```

Comment:
Where are we using this action creator?

Reply:
It isn't used anywhere. I've removed it ✅